### PR TITLE
chore: added warning message when user type changed

### PR DESCRIPTION
**Pull request by meshramsaroj**
**Task Link:** https://app.asana.com/0/1194051119161697/1200367023041052/f

**Description:** Added warning message when user type is changed from website user to system user and vise versa.

**Screenshot:**

-  when user type changed from website user to system user
image

- when user type changed from system user to website user
image

### CODE REVIEW

File: frappe/core/doctype/user/user.py [hunk 1]
```diff
@@ -513,27 +519,6 @@ def get_restricted_ip_list(self):
 
 		return [i.strip() for i in self.restrict_ip.split(",")]
 
-	@classmethod
-	def find_by_credentials(cls, user_name, password, validate_password=True):
```
Comment (hrwX) [inline frappe/core/doctype/user/user.py:517]: Why did we remove this ?

**Review by hrwX — CHANGES_REQUESTED** [review]

File: frappe/core/doctype/user/user.py (hunk 1 above)
Comment (meshramsaroj) [reply frappe/core/doctype/user/user.py:517]: this code has been removed while checking out to the new branch.
now I restore removed code

**Review by hrwX — DISMISSED** [review]

File: frappe/core/doctype/user/user.py [hunk 2]
```diff
@@ -160,10 +160,16 @@ def email_new_password(self, new_password=None):
 
 	def set_system_user(self):
 		'''Set as System User if any of the given roles has desk_access'''
+		old_user_type = self.user_type
+
 		if self.has_desk_access() or self.name == 'Administrator':
 			self.user_type = 'System User'
+			if self.user_type != old_user_type:
+				msgprint(_("User Type changed from {0} to {1}").format(old_user_type, self.user_type), title=_('Warning'), indicator='red')
```
Comment (sahil28297) [inline frappe/core/doctype/user/user.py]:
```suggestion
```

File: frappe/core/doctype/user/user.py [hunk 3]
```diff
@@ -160,10 +160,16 @@ def email_new_password(self, new_password=None):
...
 		else:
 			self.user_type = 'Website User'
+			if self.user_type != old_user_type:
+				msgprint(_("User Type changed from {0} to {1}").format(old_user_type, self.user_type), title=_('Warning'), indicator='red')
```
Comment (sahil28297) [inline frappe/core/doctype/user/user.py]:
```suggestion

		if self.user_type != old_user_type:
			msgprint(_("User Type changed from {0} to {1}").format(old_user_type, self.user_type), title=_('Warning'), indicator='red')
```

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: frappe/core/doctype/user/user.py (hunk 3 above)
Comment (meshramsaroj) [reply frappe/core/doctype/user/user.py]: resolved

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**